### ISSUE TITLE
recursive include path isn't working

### ISSUE THREAD

**Issue by byenghee** [microsoft/vscode-cpptools#8807]
Bug type: Language Service

**Describe the bug**
- OS and Version: Windows 10 Enterprise 20H2
- VS Code Version: 1.64.1
- C/C++ Extension Version: 1.8.4
- Other extensions you installed (and if the issue persists after disabling them):
- If using SSH remote, specify OS of remote machine: No
- Intellisense "includePath"'s  recursive path option not working.
even I set it with ${workspaceFolder}/**, cpp fail to find header under sub folder.

**Steps to reproduce**

1.  make simple work space with sub folder with header file like following example
main.cpp
#include "test.h"
int main()
{
    test1();
    printf("end\n");
}

inc/test.h
#pragma once
void test1();

2. set c/c++ intellisense include path with ${workspaceFolder}/**
3. check main.cpp file if intellisence find header.
4. I got below error my env.
#include errors detected. Please update your includePath. Squiggles are disabled for this translation unit (C:\Work\test\main.cpp).
cannot open source file "test.h"

**Expected behavior**

main.cpp should find header path with ${workspaceFolder}/** setting
when I put exact path like ${workspaceFolder}/inc than ${workspaceFolder}/**, error is fixed. 
all recurssive path setting seem not to work.

**Code sample and logs**

- Code sample
above mentioned.

- Configurations in `c_cpp_properties.json`
{
    "configurations": [
        {
            "name": "Win32",
            "includePath": [
                "${workspaceFolder}/**",
            ],
            "defines": [
                "_DEBUG",
                "UNICODE",
                "_UNICODE"
            ],
            "windowsSdkVersion": "10.0.17763.0",
            "compilerPath": "C:/Program Files (x86)/Microsoft Visual Studio/2017/Professional/VC/Tools/MSVC/14.16.27023/bin/Hostx64/x64/cl.exe",
            "cStandard": "c17",
            "cppStandard": "c++17",
            "intelliSenseMode": "windows-msvc-x64"
        }
    ],
    "version": 4
}

- Logs from running `C/C++: Log Diagnostics` from the VS Code command palette
-------- Diagnostics - 08/02/2022, 20:48:55
Version: 1.8.4
Current Configuration:
{
    "name": "Win32",
    "includePath": [
        "${workspaceFolder}/**"
    ],
    "defines": [
        "_DEBUG",
        "UNICODE",
        "_UNICODE"
    ],
    "windowsSdkVersion": "10.0.17763.0",
    "compilerPath": "C:/Program Files (x86)/Microsoft Visual Studio/2017/Professional/VC/Tools/MSVC/14.16.27023/bin/Hostx64/x64/cl.exe",
    "cStandard": "c17",
    "cppStandard": "c++17",
    "intelliSenseMode": "windows-msvc-x64",
    "compilerPathIsExplicit": true,
    "cStandardIsExplicit": true,
    "cppStandardIsExplicit": true,
    "intelliSenseModeIsExplicit": true,
    "compilerArgs": [],
    "mergeConfigurations": false,
    "browse": {
        "path": [],
        "limitSymbolsToIncludedHeaders": true
    }
}
Translation Unit Mappings:
[ C:\Work\test\main.cpp ]:
    C:\Work\test\main.cpp
Translation Unit Configurations:
[ C:\Work\test\main.cpp ]:
    Process ID: 38992
    Memory Usage: 35 MB
    Compiler Path: C:/Program Files (x86)/Microsoft Visual Studio/2017/Professional/VC/Tools/MSVC/14.16.27023/bin/Hostx64/x64/cl.exe
    Includes:
        C:\PROGRAM FILES (X86)\MICROSOFT VISUAL STUDIO\2017\PROFESSIONAL\VC\TOOLS\MSVC\14.16.27023\INCLUDE
        C:\PROGRAM FILES (X86)\MICROSOFT VISUAL STUDIO\2017\PROFESSIONAL\VC\TOOLS\MSVC\14.16.27023\ATLMFC\INCLUDE
        C:\PROGRAM FILES (X86)\WINDOWS KITS\10\INCLUDE\10.0.17763.0\UM
        C:\PROGRAM FILES (X86)\WINDOWS KITS\10\INCLUDE\10.0.17763.0\UCRT
        C:\PROGRAM FILES (X86)\WINDOWS KITS\10\INCLUDE\10.0.17763.0\SHARED
        C:\PROGRAM FILES (X86)\WINDOWS KITS\10\INCLUDE\10.0.17763.0\WINRT
        C:\PROGRAM FILES (X86)\WINDOWS KITS\10\INCLUDE\10.0.17763.0\CPPWINRT
    Defines:
        _DEBUG
        UNICODE
        _UNICODE
    Standard Version: ms_c++17
    IntelliSense Mode: windows-msvc-x64
Total Memory Usage: 35 MB

------- Workspace parsing diagnostics -------
Number of files discovered (not excluded): 4739

- Logs from [the language server logging](https://code.visualstudio.com/docs/cpp/enable-logging-cpp#_enable-logging-for-the-language-server)

**Screenshots**

**Additional context**

**Comment by sean-mcmanus** [comment]
It's working for me -- can you try running the C/C++: Reset IntelliSense Database command? And if that still doesn't fix it, can you set C_Cpp.loggingLevel to "Debug" and run the Reset again to look for tag parsing messages for headers under the recursive include path that is expected to be added?

**Comment by byenghee** [comment]
it didn't fix issue. below is log after reset intellisense db.
any other idea?
``
loggingLevel: Debug
loggingLevel has changed to: Debug
cpptools/didChangeCppProperties
$/setTraceNotification
cpptools/getCodeActions: C:\Work\test\main.cpp (id: 122)
cpptools/textEditorSelectionChange
cpptools/textEditorSelectionChange
cpptools/activeDocumentChange: C:\Work\test\main.cpp
cpptools/getFoldingRanges: C:\Work\test\main.cpp (id: 123)
cpptools/getCodeActions: C:\Work\test\main.cpp (id: 124)
cpptools/resetDatabase
cpptools/resetDatabaseReady
Database safe to open
Code browsing service initialized
Discovering files...
  Processing folder (non-recursive): C:/PROGRAM FILES (X86)/MICROSOFT VISUAL STUDIO/2017/PROFESSIONAL/VC/TOOLS/MSVC/14.16.27023/ATLMFC/INCLUDE
  Processing folder (non-recursive): C:/PROGRAM FILES (X86)/MICROSOFT VISUAL STUDIO/2017/PROFESSIONAL/VC/TOOLS/MSVC/14.16.27023/INCLUDE
  Processing folder (recursive): C:/PROGRAM FILES (X86)/WINDOWS KITS/10/INCLUDE/10.0.17763.0/CPPWINRT/
  Processing folder (recursive): C:/PROGRAM FILES (X86)/WINDOWS KITS/10/INCLUDE/10.0.17763.0/SHARED/
  Processing folder (recursive): C:/PROGRAM FILES (X86)/WINDOWS KITS/10/INCLUDE/10.0.17763.0/UCRT/
  Processing folder (recursive): C:/PROGRAM FILES (X86)/WINDOWS KITS/10/INCLUDE/10.0.17763.0/UM/
  Processing folder (recursive): C:/PROGRAM FILES (X86)/WINDOWS KITS/10/INCLUDE/10.0.17763.0/WINRT/
  Discovering files: 4739 file(s) processed
  0 file(s) removed from database
Done discovering files.
Populating include completion cache.
Parsing remaining files...
  Parsing: 0 files(s) processed
Done parsing remaining files.
cpptools/getCodeActions: C:\Work\test\main.cpp (id: 125)
0 include path suggestion(s) discovered.
cpptools/textEditorSelectionChange
cpptools/activeDocumentChange: C:\Work\test\main.cpp

``

`
PS C:\Work\test> ls


    Directory: C:\Work\test


Mode                 LastWriteTime         Length Name                                                                                                                                                                                                                                                    
----                 -------------         ------ ----                                                                                                                                                                                                                                                    
d-----        09/02/2022     09:54                .vscode                                                                                                                                                                                                                                                 
d-----        08/02/2022     19:41                inc                                                                                                                                                                                                                                                     
-a----        08/02/2022     19:36            123 main.cpp                                                                                                                                                                                                                                                
-a----        08/02/2022     19:39             81 test.cpp                                                                                                                                                                                                                                                


PS C:\Work\test>
`

**Comment by sean-mcmanus** [comment]
Something really strange is occurring. In both cases, your logging should show `"Processing folder (recursive): <workspace folder>`, i.e. that's probably the root cause of the failure (and would also cause other features to fail that required parsing of global symbols of source files in that folder, unless those files were manually opened). If you don't see that logging, then files won't get processed from that location, which would subsequently prevent the recursive includes feature from working correctly. Do you have any files.exclude or C_Cpp.files.exclude that could be causing the workspace folder to be excluded?

**Comment by byenghee** [comment]
no. I don't have any exclude setting on my setup.
I tried old version of vs code and c++ intellisense extension for the case.
what I found that I saw issue from c++ intellisense extension v 1.8.2 + any version of VS code.
the issue fixed when I back to intellisense extension v 1.7.1 ( 3 month ago version)
I used same work space without any change, only switch version of extensions.
I would stick to use old version extension for now, but can you check this from your side.
I am pretty sure this issue isn't only for me.

**Comment by sean-mcmanus** [comment]
This appears to be the same issue as https://github.com/microsoft/vscode-cpptools/issues/8851 -- it's caused by tag parsing failing (the message `Tag parsing encountered an error, but it may not matter. Let us know if symbols in the file can't be found: C:\WORK\TEST\MAIN.CPP` in your logging).

**Comment by sean-mcmanus** [comment]
We've added tag parser error logging with 1.9.3 (Pre-Release) with C_Cpp.loggingLevel set to "Warning" or greater. You should see some text like
```
  tag parsing file: <path>
    tag parsing error (this can be ignored unless symbols can't be found):
    line 8 (col. 8): error: incorrectly formed universal character name
```
in the C/C++ logging pane. If you try that and let us know what error message is being shown that could help us fix the issue.

**Comment by sean-mcmanus** [comment]
This should be fixed with 1.9.4 (pre-release).

**Comment by byenghee** [comment]
yes, I confirm that the issue fixed with 1.9.4. thanks for help.
